Pokedex/Pokedex/Presentation/Pokemon/JRMPokemonTableViewCell.m

```diff
@@ -34,5 +37,60 @@ -(void)setPokemon:(JRMPokemon *)pokemon{
     _pokemon = pokemon;
     self.lblName.text = pokemon.name;
     [self.ivSprite sd_setImageWithURL:[NSURL URLWithString:pokemon.sprites.frontDefault]];
+    NSMutableArray *colors = [NSMutableArray arrayWithObject:FlatWhite];
+    for (JRMType *type in pokemon.types) {
+        NSString* typeStr =  type.type[@"name"]; // Yes, this API is kinda crazy...
+        [colors addObject: [self getColorForPokemonType:typeStr]];
+    }
+    self.backgroundColor = [UIColor colorWithGradientStyle:UIGradientStyleLeftToRight
+                                                 withFrame:self.frame
+                                                 andColors:colors];
+    self.lblName.textColor = [UIColor colorWithContrastingBlackOrWhiteColorOn:self.backgroundColor isFlat:NO];
+}
+
```

Comment:
This method has a couple of flaws:
- it is too big due to excessive conditions
- hardcoded values
- inside a cell probably isn't the perfect place to define this behaviour

consider refactoring it to another file, a JRMPokemon category perharps, take a look on the following suggestion:
- Create a JRMPokemon+Color file, and inside it add this...
```objective-c

#define POKEMON_TYPE_NORMAL @"normal"
#define POKEMON_TYPE_FIGHTING @"fighting"
...

-(NSArray<UIColor *>*) colorsForTypes{
    
    NSMutableArray* colors = [NSMutableArray array];
    NSDictionary *colorsForTypes = @{ POKEMON_TYPE_NORMAL : FlatWhiteDark,
                                      POKEMON_TYPE_FIGHTING : FlatBrownDark,
                                      ...
                                      POKEMON_TYPE_SHADOW : FlatNavyBlue};
    
    for (JRMType *type in self.types) {
        NSString* typeStr =  type.type[@"name"]; // Yes, this API is kinda crazy...
        UIColor * colorForType = colorsForTypes[typeStr];
        if (colorForType != nil){
            [colors addObject: colorForType];
        }
    }
    
    return colors;
}
```